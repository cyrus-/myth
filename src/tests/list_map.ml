let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let zero (n:nat) : nat = O ;;
let inc (n:nat): nat = S (n) ;;

let list_map : (nat -> nat) -> list -> list |>
  { inc => ( [] => []
            | [0] => [1]
            | [0; 0] => [1; 1]
            | [1] => [2]
            | [1; 1] => [2; 2] )
  | zero => ( [] => []
            | [0] => [0]
            | [0; 0] => [0; 0] )
  } = ?"
