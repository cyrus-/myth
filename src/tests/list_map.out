(* Generated Thu Nov 13 00:35:27 2014 *)
let list_map : (nat -> nat) -> list -> list =
  fun (f2:nat -> nat) ->
    let rec f3 (l1:list) : list =
      match l1 with
        | Nil -> Nil
        | Cons (n1, l2) -> Cons (f2 n1, f3 l2)
    in
      f3
;;
